packages/amplify-graphql-api-construct/src/amplify-graphql-api.ts

```diff
@@ -401,7 +396,7 @@ export class AmplifyGraphqlApi extends Construct {
  * @param scope the scope this construct is created in.
  */
 const validateNoOtherAmplifyGraphqlApiInStack = (scope: Construct): void => {
-  const rootStack = getStackForScope(scope, true);
+  const rootStack = getStackForScope(scope, false);
```

Comment:
The second argument here is to check the whole tree (when true) or to only check the parent (when false).

https://github.com/aws-amplify/amplify-category-api/blob/a5a233cfc1aa0859a9adf74f319b53c9479b38d4/packages/amplify-graphql-api-construct/src/internal/construct-tree.ts#L9-L16